### PR TITLE
update go sample to declarative function and add multiple functions sample

### PR DESCRIPTION
**Pull request by lizzzcai**
follow up PR of https://github.com/OpenFunction/functions-framework-go/pull/48

Need to release the builder `openfunction/builder-go:latest` (https://github.com/OpenFunction/builder/pull/56) before merging this.

### CODE REVIEW

File: functions/knative/hello-world-go/go.mod [hunk 1]
```diff
@@ -1,3 +1,5 @@
 module example.com/hello
 
 go 1.16
+
+require github.com/OpenFunction/functions-framework-go v0.2.4
```
Comment (benjaminhuo) [inline functions/knative/hello-world-go/go.mod]: 0.3.0?

File: functions/knative/hello-world-go/README.md [hunk 2]
```diff
@@ -6,6 +6,27 @@
 
 You can refer to the [Installation Guide](https://github.com/OpenFunction/OpenFunction#install-openfunction) to setup OpenFunction.
 
+## Run it locally
+
+Build the function locally
+
+```sh
+pack build func-helloworld-go --builder openfunction/builder-go:v2.3.0 --env FUNC_NAME="HelloWorld"  --env FUNC_CLEAR_SOURCE=true
```
Comment (benjaminhuo) [inline functions/knative/hello-world-go/README.md:14]: maybe the builder should have the go version as a suffix:
builder-go:v2.3.0-1.16

Comment (lizzzcai) [reply functions/knative/hello-world-go/README.md]: done. by default we can create tag `v2.3.0` to the latest builder with the latest supported go version as well.

**Comment by benjaminhuo** [comment]
I'll merge this once the builder is pushed to openfunction dockerhub
@lizzzcai let me know your private dockerhub once the builder is pushed to it, I'll sync it to openfunction's

**Comment by lizzzcai** [comment]
> I'll merge this once the builder is pushed to openfunction dockerhub @lizzzcai let me know your private dockerhub once the builder is pushed to it, I'll sync it to openfunction's

```
lizzzcai/buildpacks-run-go:v2.3.0-1.16 -> openfunctiondev/buildpacks-run-go:v2.3.0-1.16
lizzzcai/buildpacks-go116-build:v2.3.0-1.16 -> openfunctiondev/buildpacks-go116-build:v2.3.0-1.16
lizzzcai/builder-go:v2.3.0-1.16 -> openfunction/builder-go:v2.3.0-1.16
```

**Comment by benjaminhuo** [comment]
image
builder uploaded including the new latest image, thanks @lizzzcai

**Pull request merged**